src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -562,3 +563,21 @@ zocl_read_axlf_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 	vfree(axlf);
 	return ret;
 }
+
+int
+zocl_info_cu_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
+{
+	struct drm_zocl_info_cu *args = data;
+	struct drm_zocl_dev *zdev = dev->dev_private;
+	struct sched_exec_core *exec = zdev->exec;
+
+	if (!exec->configured) {
+		DRM_ERROR("Schduler is not configured\n");
+		return -EINVAL;
+	}
+ 
+	if(args->paddr != 0xFFFFFFFF) {
+		args->apt_idx = get_apt_index(zdev, args->paddr);
+	}
```

Comment:
if args->paddr does equal to 0xFFFFFFFF, this function will return 0 and apt_idx is not defined. Is it what expected here?
I think the better way is check the return value of get_apt_index, if it is -EINVAL, just return -EINVAL of your ioctl function. So that the shim.cpp code (ioctl caller) does not need to check if the apt_index is -EINVAL. And if we have other failuer case in kernel, just return error in ioctl handler and ioctl caller does not need to add extra check

Reply:
code updated